sdk/python/kfp/components/types/type_annotations.py

```diff
@@ -252,3 +254,11 @@ def get_short_type_name(type_name: str) -> str:
         return match.group('type')
     else:
         return type_name
+
+
+def is_artifact(
+        artifact_class_or_instance: Union[type,
+                                          artifact_types.Artifact]) -> bool:
+    # we do not yet support non-pre-registered custom artifact types with instance_schema attribute
+    return hasattr(artifact_class_or_instance, 'schema_title') and hasattr(
+        artifact_class_or_instance, 'schema_version')
```

Comment:
This is our new definition for what an artifact is, that way non-KFP libraries can implement an artifact class that adheres to the KFP artifact protocol without having a dependency on KFP.